rust-runtime/aws-smithy-http-server/src/proto/mod.rs

```diff
@@ -0,0 +1,7 @@
+/*
```

Comment:
Thinking out loud: Should we flatten this a bit and take `rest_json_1`/`rest_xml` out of `rest` and put them at this level? The common structs would still live in `rest` - the `rest` folder wouldn't contain a protocol ZST, but will contain a `router` module etc.

Reply:
Couple this with https://github.com/awslabs/smithy-rs/pull/1753#discussion_r976310182 and everything might be more predictable for codegen - `proto/{protocol}/router.rs` always contains the `Router` and `Error` for every protocol.

Reply:
```
/proto/rest/router/RestRouter
/proto/rest/router/Error
/proto/rest_json_1/RestJson1
/proto/rest_json_1/router/RestRouter (re-exported)
/proto/rest_json_1/router/Error (re-exported)
/proto/rest_xml/RestXml
/proto/rest_xml/router/RestRouter (re-exported)
/proto/rest_xml/router/Error (re-exported)
/proto/aws_json/...
/proto/aws_json_1/...
```